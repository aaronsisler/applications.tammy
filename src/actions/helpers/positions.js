export const setPositions = (positions) => ({
    type: 'SET_POSITIONS',
    positions
});
