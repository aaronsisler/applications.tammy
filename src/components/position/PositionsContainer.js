import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import selectPositions from 'Selectors/positions';
import PositionDetails from 'Position/PositionDetails';
import PositionsList from 'Shared/position/PositionsList';
import PositionsListFilter from 'Position/PositionsListFilter';

export class PositionsContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    handleMobileClassname = () => this.props.positionId ? "inbox_mobile" : "";

    render() {
        return (
            <div className="inbox_container">
                {this.props.positions &&
                    <div className="inbox_widget">
                        <div className={`inbox_list ${this.handleMobileClassname()}`}>
                            <PositionsListFilter />
                            <PositionsList positions={this.props.positions} linkRoute="positions" />
                        </div>
                        <PositionDetails {...this.props} />
                    </div>
                }
            </div>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = (state, props) => ({
    positionId: props.match.params.positionId,
    positions: selectPositions(state.positions, state.filters.positions),
});

export default connect(mapStateToProps)(PositionsContainer);

PositionsContainer.propTypes = {
    positionId: PropTypes.string,
    positions: PropTypes.array,
};
