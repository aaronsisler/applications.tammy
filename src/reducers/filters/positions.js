const positionsFilterReducerDefaultState =
{
    text: '',
};

const positionsFilterReducer = (state = positionsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'CLEAR_POSITIONS_TEXT_FILTER':
            return {
                ...state,
                text: ''
            };
        default:
            return state;
    }
};

export default positionsFilterReducer;
