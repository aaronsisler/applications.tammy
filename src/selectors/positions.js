// Get filtered positions
export default (positions, { text = '' }) => positions
        .filter((position) => {
            const titleMatch = position.title.toLowerCase().includes(text.toLowerCase());
            const jobIdMatch = position.jobId.toLowerCase().includes(text.toLowerCase());
            return jobIdMatch || titleMatch;
        })
