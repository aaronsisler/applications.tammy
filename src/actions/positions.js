import database from '../firebase/firebase';
import { setPositions } from 'Actions/helpers/positions';

export const startSetPositions = () => (dispatch) =>
    database.ref(`positions`).once('value').then((snapshot) => {
        const positions = [];
        snapshot.forEach((childSnapshot) => {
            positions.push({
                positionId: childSnapshot.key,
                ...childSnapshot.val(),
            });
        });

        dispatch(setPositions(positions));
    });
