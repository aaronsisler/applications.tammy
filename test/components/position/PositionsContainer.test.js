import React from 'react';
import { shallow } from 'enzyme';
import { PositionsContainer } from 'Position/PositionsContainer';
import positions from '../../fixtures/positions';

describe('PositionsContainer', () => {
    let wrapper;

    const buildWrapper = (positionsInput = [], positionInput = {}, filters = { text: '' }) => {
        wrapper = shallow(
            <PositionsContainer
                filters={filters}
                position={positionInput}
                positions={positionsInput}
            />
        );
    };

    it('should render PositionsContainer correctly when a position is selected', () => {
        const [position] = positions;
        buildWrapper(positions, position);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render PositionsContainer correctly when a position is NOT selected', () => {
        buildWrapper(positions);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render PositionsContainer correctly when text filter is applied', () => {
        const [{ jobId }] = positions;
        buildWrapper(positions, undefined, { text: jobId });

        expect(wrapper).toMatchSnapshot();
    });
});
