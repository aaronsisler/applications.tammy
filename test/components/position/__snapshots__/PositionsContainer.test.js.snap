// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionsContainer should render PositionsContainer correctly when a position is NOT selected 1`] = `
<div
  className="inbox_container"
>
  <div
    className="inbox_widget"
  >
    <div
      className="inbox_list "
    >
      <Connect(PositionsListFilter) />
      <PositionsList
        linkRoute="positions"
        positions={
          Array [
            Object {
              "content": Object {
                "description": "first description",
                "requirements": Array [
                  "Skill 1",
                  "Skill 2",
                ],
                "responsibilities": Array [
                  "Responsibility 1",
                  "Responsibility 2",
                ],
              },
              "jobId": "first job id",
              "location": "first location",
              "positionId": "firstPositionId",
              "title": "first title",
            },
            Object {
              "content": Object {
                "description": "second description",
                "requirements": Array [
                  "Skill 1",
                  "Skill 2",
                ],
                "responsibilities": Array [
                  "Responsibility 1",
                  "Responsibility 2",
                ],
              },
              "jobId": "second job id",
              "location": "second location",
              "positionId": "secondPositionId",
              "title": "second title",
            },
          ]
        }
      />
    </div>
    <Connect(PositionDetails)
      filters={
        Object {
          "text": "",
        }
      }
      position={Object {}}
      positions={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "positionId": "firstPositionId",
            "title": "first title",
          },
          Object {
            "content": Object {
              "description": "second description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "second job id",
            "location": "second location",
            "positionId": "secondPositionId",
            "title": "second title",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`PositionsContainer should render PositionsContainer correctly when a position is selected 1`] = `
<div
  className="inbox_container"
>
  <div
    className="inbox_widget"
  >
    <div
      className="inbox_list "
    >
      <Connect(PositionsListFilter) />
      <PositionsList
        linkRoute="positions"
        positions={
          Array [
            Object {
              "content": Object {
                "description": "first description",
                "requirements": Array [
                  "Skill 1",
                  "Skill 2",
                ],
                "responsibilities": Array [
                  "Responsibility 1",
                  "Responsibility 2",
                ],
              },
              "jobId": "first job id",
              "location": "first location",
              "positionId": "firstPositionId",
              "title": "first title",
            },
            Object {
              "content": Object {
                "description": "second description",
                "requirements": Array [
                  "Skill 1",
                  "Skill 2",
                ],
                "responsibilities": Array [
                  "Responsibility 1",
                  "Responsibility 2",
                ],
              },
              "jobId": "second job id",
              "location": "second location",
              "positionId": "secondPositionId",
              "title": "second title",
            },
          ]
        }
      />
    </div>
    <Connect(PositionDetails)
      filters={
        Object {
          "text": "",
        }
      }
      position={
        Object {
          "content": Object {
            "description": "first description",
            "requirements": Array [
              "Skill 1",
              "Skill 2",
            ],
            "responsibilities": Array [
              "Responsibility 1",
              "Responsibility 2",
            ],
          },
          "jobId": "first job id",
          "location": "first location",
          "positionId": "firstPositionId",
          "title": "first title",
        }
      }
      positions={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "positionId": "firstPositionId",
            "title": "first title",
          },
          Object {
            "content": Object {
              "description": "second description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "second job id",
            "location": "second location",
            "positionId": "secondPositionId",
            "title": "second title",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`PositionsContainer should render PositionsContainer correctly when text filter is applied 1`] = `
<div
  className="inbox_container"
>
  <div
    className="inbox_widget"
  >
    <div
      className="inbox_list "
    >
      <Connect(PositionsListFilter) />
      <PositionsList
        linkRoute="positions"
        positions={
          Array [
            Object {
              "content": Object {
                "description": "first description",
                "requirements": Array [
                  "Skill 1",
                  "Skill 2",
                ],
                "responsibilities": Array [
                  "Responsibility 1",
                  "Responsibility 2",
                ],
              },
              "jobId": "first job id",
              "location": "first location",
              "positionId": "firstPositionId",
              "title": "first title",
            },
            Object {
              "content": Object {
                "description": "second description",
                "requirements": Array [
                  "Skill 1",
                  "Skill 2",
                ],
                "responsibilities": Array [
                  "Responsibility 1",
                  "Responsibility 2",
                ],
              },
              "jobId": "second job id",
              "location": "second location",
              "positionId": "secondPositionId",
              "title": "second title",
            },
          ]
        }
      />
    </div>
    <Connect(PositionDetails)
      filters={
        Object {
          "text": "first job id",
        }
      }
      position={Object {}}
      positions={
        Array [
          Object {
            "content": Object {
              "description": "first description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "first job id",
            "location": "first location",
            "positionId": "firstPositionId",
            "title": "first title",
          },
          Object {
            "content": Object {
              "description": "second description",
              "requirements": Array [
                "Skill 1",
                "Skill 2",
              ],
              "responsibilities": Array [
                "Responsibility 1",
                "Responsibility 2",
              ],
            },
            "jobId": "second job id",
            "location": "second location",
            "positionId": "secondPositionId",
            "title": "second title",
          },
        ]
      }
    />
  </div>
</div>
`;
